src/Conversion/KrnlToLLVM/KrnlEntryPoint.cpp

```diff
@@ -159,79 +168,95 @@ class KrnlEntryPointOpLowering : public OpRewritePattern<KrnlEntryPointOp> {
             }, /*then=*/
             [&](LLVMBuilder &createLLVM) {
               // return NULL.
-              createLLVM._return(createLLVM.nullI8Ptr());
+              createLLVM._return(createLLVM.null(getI8PointerType(context)));
             });
       }
     }
 
-    // Based on the static entry point type signature, unpack dynamic memory
-    // refs to corresponding static memory refs.
-    auto wrappedStaticEntryPointFuncName =
-        "_mlir_ciface_" + staticEntryPointFuncName.lower();
-    auto *staticEntryPointFunc =
-        module.lookupSymbol(wrappedStaticEntryPointFuncName);
-    assert(staticEntryPointFunc &&
-           isa<LLVM::LLVMFuncOp>(staticEntryPointFunc) &&
-           "entry point func must exist and be an llvm func op");
-    auto staticEntryPointTy = dyn_cast<LLVM::LLVMFuncOp>(staticEntryPointFunc)
-                                  .getFunctionType()
-                                  .dyn_cast<LLVM::LLVMFunctionType>();
-
-    // Retrieve dynamic mem refs from wrapped input, and convert every one of
-    // them to static mem refs.
-    SmallVector<Value, 4> staticInputs;
-    auto wrappedInput = entryPointEntryBlock.getArgument(0);
-
-    // Emit code to verify every tensor in the wrapped input, e.g. verifying
+    // 2. Emit code to verify every tensor in the wrapped input, e.g. verifying
     // shape and data type.
     if (verifyInputTensors) {
-      StringAttr sigAttr = op->getAttrOfType<StringAttr>(
-          KrnlEntryPointOp::getSignatureAttrName());
       llvm::StringRef inSigJSON;
       std::tie(inSigJSON, std::ignore) = sigAttr.getValue().split('@');
       emitVerificationCodeForInputTensors(
-          module, rewriter, loc, apiRegistry, wrappedInput, inSigJSON);
+          module, rewriter, loc, apiRegistry, omTensorInputs, inSigJSON);
     }
 
+    // 3. Emit code to prepare MemRefs from OMTensor inputs and call
+    // `_mlir_ciface` prefixed function of the entry point.
+
+    // Based on the static entry point type signature, unpack dynamic memory
+    // refs to corresponding static memory refs.
+    // Note that, in the static entry point type signature, output type is a
+    // struct but input types are unpacked into a single list of scalar types.
+    auto *staticEntryPointFunc =
+        module.lookupSymbol(staticEntryPointFuncName.lower());
+    auto staticEntryPointFuncTy =
+        dyn_cast<LLVM::LLVMFuncOp>(staticEntryPointFunc)
+            .getFunctionType()
+            .dyn_cast<LLVM::LLVMFunctionType>();
```

Comment:
Use `cast` if you don't check the result of `dyn_cast`

```suggestion
    auto staticEntryPointFuncTy =
        cast<LLVM::LLVMFuncOp>(staticEntryPointFunc)
            .getFunctionType()
            .cast<LLVM::LLVMFunctionType>();
```